libogc/sha.c

```diff
@@ -0,0 +1,157 @@
+/*-------------------------------------------------------------
+
+sha.c -- SHA Engine
+
+Copyright (C) 2023
+Joris 'DacoTaco' Vermeylen info@dacotaco.com
+
+This software is provided 'as-is', without any express or implied
+warranty.  In no event will the authors be held liable for any
+damages arising from the use of this software.
+
+Permission is granted to anyone to use this software for any
+purpose, including commercial applications, and to alter it and
+redistribute it freely, subject to the following restrictions:
+
+1.	The origin of this software must not be misrepresented; you
+must not claim that you wrote the original software. If you use
+this software in a product, an acknowledgment in the product
+documentation would be appreciated but is not required.
+
+2.	Altered source versions must be plainly marked as such, and
+must not be misrepresented as being the original software.
+
+3.	This notice may not be removed or altered from any source
+distribution.
+
+-------------------------------------------------------------*/
+#if defined(HW_RVL)
+
+#include <string.h>
+#include "gctypes.h"
+#include "gcutil.h"
+#include "ipc.h"
+#include "sha.h"
+
+static s32 __sha_fd = -1;
+static s32 __sha_hid = -1;
+
+#define SHA_HEAPSIZE 0x50
+typedef enum 
+{ 
+	ResetContext = 0x00,
+	AddData = 0x01,
+	FinalizeHash = 0x02,
+} ShaCommand;
+
+static s32 SHA_ExecuteCommand(const ShaCommand command, sha_context* context, const void* in_data, const u32 data_size, void* out_data)
+{
+	ioctlv* params = (ioctlv*)iosAlloc(__sha_hid, sizeof(ioctlv) * 3);
```

Comment:
should there be a `NULL` check here or is it guaranteed by this point that `iosAlloc` can allocate here

Reply:
the chances of it returning NULL are slim unless youre doing some threading stuff.
youre right though, ill add the NULL check